src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -1123,7 +1126,20 @@ bool entitiesQuery
 
         /* Attributes part */
 
-        std::vector<BSONElement> queryAttrV = r.getField(ENT_ATTRS).Array();
+        std::vector<BSONElement> queryAttrV;
```

Comment:
I think this should be solved in a different way. Note that, by definition, any document in the enties collection has an attributes array, thus the try/catch should not be needed if we capture the error in a early place.

In particular, looking to the logs after executing the .test included in this PR I have found this:

```
time=2015-01-02T19:31:16.884CET | lvl=DEBUG | trans=1420223475-825-00000000001 | function=entitiesQuery | comp=Orion | msg=MongoGlobal.cpp[1108]: retrieved document: '{ $err: "Exterior shell of polygon is invalid: { type: "Polygon", coordinates: [ [ [ -70.652788, -33.603458 ], [ -70.64603200000001, -33.604231 ], [ -70.646781...", code: 16693 }'
```
Thus probably a better strategy is to check if the "$err" field exists in the `r` document just after `LM_T(LmtMongo, ("retrieved document ...`) and, in that case, return the proper error response to the user (I'd suggest a 500 Internal Error, using the text in "$err" to fill the details field). I think that is better than a "No context element found" (as in the current version in the code) which may be hidding structural problems in geoquery done at DB.

That way we would avoid Orion to crash. However, we can do even better detecting those "Exterior shell of polygon is invalid" situation before interacting with DB, fixing them if possibe or reporting the user with a 400 Bad Input error about the problem if the situation is not easy to fix. Looking to http://stackoverflow.com/questions/21870088/exterior-shell-of-polygon-is-invalid, it seems they happen when the polygon have duplicated coordinates (this seems to be easy to detect) or is an "intersection polygon" (I'm not sure if al algorithm to reorder coordinates is easy to implement). 

Maybe the 500 Internat Error solution is enough for this PR. If detecting and fixing "Exterior shell of polygon is invalid" situations is difficult, we can defer it to a tech debt issue (nicetohave?).